## ДЗ по уроку № 5 (29.06.2022)
## Теория вероятности
### Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

**Пример 1**

Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

Нормальное распределение. 
Среднее квадратическое отклонение sigma = 16. 
Выборочная средняя M = 80. 
Объем выборки n = 256. 

Надежность 0.95 - решение через две сигмы: 

Математическое ожидание должно быть в интервале
(M - 2 * sigma/sqrt(n), M + 2 * sigma/sqrt(n))

(80 - 2 * 16/16, 80 + 2 * 16/16)

Ответ: доверительный интервал (78, 82)

**Пример 2**

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [1]:
import numpy as np
from scipy import stats

X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [2]:
# уровень значимости:

alfa = 0.05

# размер выборки: 

n = 10

# среднее элементов выборки: 

X_ = X.mean()

# среднее квадратичное отклонение в выборке:

sigma = X.std(ddof=1)

In [3]:
# Табличное значение t-критерия Стьюдента для alpha и n

t = stats.t.ppf(1 - alfa / 2, df=n - 1)
t

2.2621571627409915

In [4]:
# Левая граница доверительного интервала: 
    
left = X_ - t * sigma / np.sqrt(n)
left

6.267515851415713

In [5]:
# Правая граница доверительного интервала: 
    
right = X_ + t * sigma / np.sqrt(n)
right

6.912484148584288

Ответ: доверительный интервал (6.2675, 6.9125)

**Пример 3**

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

m = 17. 
alfa = 0.05. 
n = 100. 
X_ = 17.5. 
D = 4. 

Табличное значение t-критерия Стьюдента для alpha и n: t = 1.645. 

sigma = sqrt(D)

T = (X_ - m)/(sigma/sqrt(n))

T = (17.5 - 17)/(2/10)) = 0.5 / 0.2 = 2.5

Ответ: T > t - гипотеза отвергается. 

**Пример 4**

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [6]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [7]:
# уровень значимости:

alfa = 0.01

# X_ - средний вес пачки из выборки: 

X_ = X.mean()

# sigma - среднее квадратичное отклонение в выборке:

sigma = X.std(ddof=1)

# n - размер выборки:

n = 10

In [8]:
X_

198.5

In [9]:
sigma

4.453463071962462

In [10]:
# Табличное значение t-критерия Стьюдента для alpha и n

t = stats.t.ppf(1 - alfa / 2, df=n - 1)
t

3.2498355440153697

In [11]:
# Левая граница доверительного интервала: 
    
left = X_ - t * sigma / np.sqrt(n)
left

193.92322840036354

In [12]:
# Правая граница доверительного интервала: 
    
right = X_ + t * sigma / np.sqrt(n)
right

203.07677159963646

Ответ: утверждаемый вес пачки 200 по предлагаемой гипотезе входит в доверительный интервал - гипотеза верна. 